memsql-plugin/src/main/java/io/cdap/plugin/memsql/source/MemsqlSourceConfig.java

```diff
@@ -0,0 +1,125 @@
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.plugin.memsql.source;
+
+import io.cdap.cdap.api.annotation.Description;
+import io.cdap.cdap.api.annotation.Name;
+import io.cdap.cdap.api.data.schema.Schema;
+import io.cdap.cdap.etl.api.validation.InvalidConfigPropertyException;
+import io.cdap.plugin.db.batch.config.DBSpecificSourceConfig;
+import io.cdap.plugin.memsql.MemsqlConstants;
+import io.cdap.plugin.memsql.MemsqlUtil;
+
+import java.util.List;
+import java.util.Map;
+import javax.annotation.Nullable;
+
+/**
+ * Config class for {@link MemsqlSource}.
+ */
+public class MemsqlSourceConfig extends DBSpecificSourceConfig {
+
+  @Name(MemsqlConstants.AUTO_RECONNECT)
+  @Description("Should the driver try to re-establish stale and/or dead connections")
+  @Nullable
+  public Boolean autoReconnect;
+
+  @Name(MemsqlConstants.USE_COMPRESSION)
+  @Description("Select this option for WAN connections")
+  @Nullable
+  public Boolean useCompression;
+
+  @Name(MemsqlConstants.USE_SSL)
+  @Description("Turns on SSL encryption. Connection will fail if SSL is not available")
+  @Nullable
+  public String useSSL;
+
+  @Name(MemsqlConstants.USE_ANSI_QUOTES)
+  @Description("Treats \" as an identifier quote character and not as a string quote character")
+  @Nullable
+  public Boolean useAnsiQuotes;
+
+  @Name(MemsqlConstants.KEY_STORE)
+  @Description("URL to the client certificate KeyStore (if not specified, use defaults)")
+  @Nullable
+  public String keyStore;
+
+  @Name(MemsqlConstants.KEY_STORE_PASSWORD)
+  @Description("Password for the client certificates KeyStore")
+  @Nullable
+  public String keyStorePassword;
+
+  @Name(MemsqlConstants.TRUST_STORE)
+  @Description("URL to the trusted root certificate KeyStore (if not specified, use defaults)")
+  @Nullable
+  public String trustStore;
+
+  @Name(MemsqlConstants.TRUST_STORE_PASSWORD)
+  @Description("Password for the trusted root certificates KeyStore")
+  @Nullable
+  public String trustStorePassword;
+
+  @Override
+  public String getConnectionString() {
+    return MemsqlUtil.getConnectionString(host, port, database);
+  }
+
+  @Override
+  public Map<String, String> getDBSpecificArguments() {
+    return MemsqlUtil.composeDbSpecificArgumentsMap(autoReconnect, useCompression, useSSL,
+                                                    keyStore,
+                                                    keyStorePassword,
+                                                    trustStore,
+                                                    trustStorePassword);
+  }
+
+  @Override
+  public List<String> getInitQueries() {
+    return MemsqlUtil.composeDbInitQueries(useAnsiQuotes);
+  }
+
+  @Override
+  protected void validateSchema(Schema actualSchema) {
+    Schema configSchema = getSchema();
+    if (configSchema == null) {
+      throw new InvalidConfigPropertyException("Schema should not be null or empty", SCHEMA);
```

Comment:
this should be using the new FailureCollector instead of throwing these exceptions.